app/services/authService.js

```diff
@@ -115,6 +117,53 @@ class AuthService {
     };
   }
 
+  async sendActivateEmail(userEmail, userIp) {
+    const oUser = await PortalUserModel.findOne({email: userEmail}).lean();
+    if (!oUser) {
+      throw new CustomError(STATUS_TYPE.PORTAL_USER_NOT_FOUND);
+    }
+    if (oUser.is_activated) {
+      throw new CustomError(STATUS_TYPE.PORTAL_USER_ALREADY_ACTIVATED);
+    }
+
+    // check whether already sent email in 5 mins（all the 'code' type tokens are used to verify, so check the generate time of the token）
+    const existToken = await PortalTokenModel.findOne({user_id: oUser._id, type: 'code'})
+    if (existToken) {
+      if ((+new Date() - existToken.last_access_time) / 1000 < config.minEmailSendInterval) {
+        throw new CustomError(STATUS_TYPE.PORTAL_EMAIL_SEND_LIMIT);
+      } else{
+        await PortalTokenModel.deleteOne({_id: existToken._id}); // delete exist token and then send a new one.
+      }
+    }
+
+    // generate Token
+    const code = await this._getCode(oUser, userIp);
+    const siteName = config.siteName;
+    const siteUrl = config.siteUrl;
+    const activateUrl = `${siteUrl}/active-confirm/${code}`;
+    const displayName = oUser.nick_name;
+    const welcomMailPath = path.resolve(__dirname,'../views/welcome_mail.html');
+
+    const html = await ejs.renderFile( welcomMailPath, { siteName, displayName, activateUrl, siteUrl });
+    const activeEmail = {
+      async: false,
+      to: [oUser.email],
+      subject: `Welcome to Register at ${siteName}`,
+      text: `Hello ${displayName}, welcome to register. Please click the following link to activate your account: ${activateUrl}`,
+      html,
+    };
+
+    const sendEmailRes = await EmailService.sendEmail(activeEmail);
+    if (sendEmailRes.emailStatus == 0) {
```

Comment:
What does '0' stand for? Better using a constant value or enum to define magic numbers or string without any meaning for the first look.

This will make the code more meaningful and readable.